default_plugins/consoleme_default_plugins/plugins/aws/aws.py

```diff
@@ -190,42 +194,60 @@ async def fetch_iam_role(
                 tasks = []
                 role_name = role_arn.split("/")[-1]
                 # Instantiate a cached CloudAux client
-                client = await sync_to_async(boto3_cached_conn)(
-                    "iam",
-                    account_number=account_id,
-                    assume_role=config.get("policies.role_name"),
-                    retry_max_attempts=2,
-                )
+                if run_sync:
```

Comment:
This would be easier to follow if you split the sync and async flows out into separate functions that are called from this one, e.g.:

```python
if run_sync:
    role = _get_role_sync(account_id, role_name, ...)
else:
    role = await _get_role_async(blah...)
```